src/Application/Asset/StartUploadService.php

```
@@ -0,0 +1,314 @@
+        } catch (\Throwable $e) {
+            // Ensure we do not leave a pending orphaned asset when storage fails.
+            try {
+                $asset->markFailed();
+                $this->assets->save($asset);
+            } catch (\Throwable $suppressed) {
+                // If saving the failed state also fails, suppress secondary errors
+                // to preserve the original failure as the root cause.
+                unset($suppressed);
+            }
+
+            throw StorageUnavailableException::forReason('Storage adapter failure', $e);
```

PHPMD (2.15.0)

[error] Avoid using static access to class '\App\Domain\Asset\Exception\StorageUnavailableException' in method 'initiateUpload'. (undefined)

(StaticAccess)